src/migration/1726149087071-ContentPreference.ts

```diff
@@ -0,0 +1,45 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class ContentPreference1726149087071 implements MigrationInterface {
+  name = 'ContentPreference1726149087071';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "content_preference" ("referenceId" text NOT NULL, "userId" character varying NOT NULL, "type" text NOT NULL, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "status" text NOT NULL, "referenceUserId" text, CONSTRAINT "PK_3f50987656b7555fa8195d6f05b" PRIMARY KEY ("referenceId", "userId"))`,
```

Comment:
Shouldn't referenceId be uniquely indexed?

Reply:
referenceId is generic field, referenceUserId is the one which has FK, same as NotificationPreference

this is because you can't have FK to multiple tables